Public/Src/Engine/UnitTests/Processes.Detours/SandboxedProcessInfoTest.cs

```diff
@@ -0,0 +1,140 @@
+{
+    public class SandboxedProcessInfoTest : XunitBuildXLTest
+    {
+        public SandboxedProcessInfoTest(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public void SerializeSandboxedProcessInfo()
+        {
+            var pt = new PathTable();
+            var fam =
+                new FileAccessManifest(pt, CreateDirectoryTranslator())
+                {
+                    FailUnexpectedFileAccesses = false,
+                    IgnoreCodeCoverage = false,
+                    ReportFileAccesses = false,
+                    ReportUnexpectedFileAccesses = false,
+                    MonitorChildProcesses = false
+                };
+
+            var vac = new ValidationDataCreator(fam, pt);
+            vac.AddScope(A("C", "Users", "AppData"), FileAccessPolicy.AllowAll);
+            vac.AddPath(A("C", "Source", "source.txt"), FileAccessPolicy.AllowReadAlways);
+            vac.AddPath(A("C", "Out", "out.txt"), FileAccessPolicy.AllowAll);
+
+            var standardFiles = new SandboxedProcessStandardFiles(A("C", "pip", "pip.out"), A("C", "pip", "pip.err"));
+            var envVars = new Dictionary<string, string>()
+            {
+                ["Var1"] = "Val1",
+                ["Var2"] = "Val2",
+            };
+            IBuildParameters buildParameters = BuildParameters.GetFactory().PopulateFromDictionary(envVars);
+
+            SandboxedProcessInfo info = new SandboxedProcessInfo(
+                pt,
+                new StandardFileStorage(standardFiles),
+                A("C", "tool", "tool.exe"),
+                fam,
+                true,
+                null)
+            {
+                Arguments = @"/arg1:val1 /arg2:val2",
+                WorkingDirectory = A("C", "Source"),
+                EnvironmentVariables = buildParameters,
+                Timeout = TimeSpan.FromMinutes(10),
+                PipSemiStableHash = 0x12345678,
+                PipDescription = nameof(SerializeSandboxedProcessInfo),
+                ProcessIdListener = null,
+                TimeoutDumpDirectory = A("C", "Timeout"),
+                SandboxKind = global::BuildXL.Utilities.Configuration.SandboxKind.Default,
+                AllowedSurvivingChildProcessNames = new[] { "conhost.exe", "mspdbsrv.exe" },
+                NestedProcessTerminationTimeout = SandboxedProcessInfo.DefaultNestedProcessTerminationTimeout,
+                StandardInputSourceInfo = SandboxedProcessInfo.StandardInputInfo.CreateForData("Data"),
+                StandardObserverDescriptor = new SandboxedProcessInfo.ObserverDescriptor()
+                {
+                    WarningRegex = new SandboxedProcessInfo.RegexDescriptor("*warn", System.Text.RegularExpressions.RegexOptions.Compiled)
+                },
+            };
+
+            // Serialize and deserialize.
+            string file = Path.GetTempFileName();
+
+            using (FileStream fileStream = File.OpenWrite(file))

```

Comment:
>FileStream fileStream = File.OpenWrite(file) [](start = 19, length = 44)

Any reason to not just use a memorystream to save the I/O?

Reply:
No particular reason. I'm changing it to memory stream.

---
In reply to: [276482723](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276482723) [](ancestors = 276482723)

Reply:
Still a FileStream. Reactivating in case you forgot

---
In reply to: [276867353](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276867353) [](ancestors = 276867353,276482723)

Reply:
I fixed one in `FileAccessManifest.cs`, but apparently your comment here is in a different file :-)

---
In reply to: [277861318](https://github.com/Microsoft/BuildXL/pull/159#discussion_r277861318) [](ancestors = 277861318,276867353,276482723)